src/dmd/expression.d

```diff
@@ -202,15 +202,23 @@ Lagain:
     else
     {
         if (!s.isFuncDeclaration()) // functions are checked after overloading
+        {
             s.checkDeprecated(loc, sc);
+            if (d)
+                d.checkDisabled(loc, sc);
```

Comment:
Couldn't you do `s.checkDisabled(...)` instead, without having to check `d`?

Reply:
`checkDisabled` is a method of the `Declaration` class.  `s` is a `DSymbol` class.  Therefore, `s` must first be converted into `d` with the `isDeclaration` function.

Reply:
You could add the function into DSymbol, then override it in the Declaration class. It'd be cleaner then having all these if statements everywhere.

Reply:
That is correct, but it would depart from the patterns DMD is currently using.  grep the source code for `isDeclaration` and many other `isXXX` functions and you'll see.  Also, if I make such systematic changes to this PR, it will turn into a monstrosity and noone will want to review it.  Such a change could be a candidate for a different PR. though.